cluster/addons/dns/skydns-rc.yaml.in

```diff
@@ -19,6 +19,9 @@ spec:
         version: v11
         kubernetes.io/cluster-service: "true"
     spec:
+{% if grains['cloud'] is defined and grains['cloud'] in [ 'vsphere', 'photon-controller' ] %}
```

Comment:
The general answer is that pods don't assume.  There are places where they
don't need to escape their cluster and places where they will.  I expect a
lot of OSS projects *will* want some way to get internet access, but that
becomes a policy thing.

That it doesn't work at all -- CAN'T be made to work it seems -- makes me a
little nervous about overall utility of the solution, but that's not on me
to judge.  If your solution demands that pods can not access anything
outside of themselves, then there's little need for an upstream DNS at
all...

On Mon, May 2, 2016 at 9:30 AM, Alain Roy <notifications@github.com> wrote:

> In cluster/addons/dns/skydns-rc.yaml.in
> <https://github.com/kubernetes/kubernetes/pull/24124#discussion_r61762838>
> :
>
> > @@ -19,6 +19,9 @@ spec:
> >          version: v11
> >          kubernetes.io/cluster-service: "true"
> >      spec:
> > +{% if grains['cloud'] is defined and grains['cloud'] in [ 'vsphere', 'photon-controller' ] %}
>
> I don't think our users will have different requirements from other
> Kubernetes users. Are other users okay with it not working with upstream
> DNS servers?
>
> What is the general expectation for pods? Should they be able to reach
> external networks (DNS or otherwise) without using the hostNetwork option?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24124/files/5699d51efadb41b6c010f45672c4844d84de0b84#r61762838>
>